lib/appsignal/cli/diagnose/paths.rb

```diff
@@ -84,9 +86,26 @@ def path_stat(path)
         end
 
         # Returns the AppSignal gem installation path. The root directory of
-        # this gem.
+        # this gem when installed.
         def gem_path
-          File.expand_path("../../../..", __dir__)
+          gemspec.full_gem_path
+        end
+
+        # Returns the AppSignal gem's Makefile log path, if it exists.
+        def makefile_install_log_path
+          possible_locations = [
+            # Installed gem location
+            File.join(gemspec.extension_dir, "mkmf.log"),
+            # Local development location
+            File.join(gem_path, "ext", "mkmf.log")
+          ]
+          possible_locations.find do |location|
+            location if File.exist?(location)
+          end
```

Comment:
A thought, feel free to ignore: instead of returning `nil` as the path when neither of the paths exist, we could return `possible_locations.last` and avoid the path potentially being `nil` (and having to check for that) at other points (in `path_stat` above and in the [diagnose tests PR](https://github.com/appsignal/diagnose_tests/pull/105))

```suggestion
          end || possible_locations.last
```